javaparser-core/src/main/java/com/github/javaparser/ast/body/EnumDeclaration.java

```diff
@@ -140,7 +141,7 @@ public EnumDeclaration setImplementedTypes(NodeList<ClassOrInterfaceType> implem
     }
 
     public EnumConstantDeclaration addEnumConstant(String name) {
-        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(assertNotNull(name));
+        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(name);
```

Comment:
Yeah I struggled with that decision. Double check or confuse the user... As the JIT will ignore the second check I think your right. I will fix some more places.